ddtrace/tracer/post_processor.go

```diff
@@ -0,0 +1,188 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package tracer
+
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var _ ddtrace.ReadWriteSpan = (*readWriteSpan)(nil)
+
+// readWriteSpan wraps span and implements the ddtrace.ReadWriteSpan interface.
+type readWriteSpan struct {
+	*span
+}
+
+// Tag returns the tag value held by the given key.
+func (s readWriteSpan) Tag(key string) interface{} {
+	s.Lock()
+	defer s.Unlock()
+
+	switch key {
+	// String.
+	case ext.SpanName:
+		return s.Name
+	case ext.ServiceName:
+		return s.Service
+	case ext.ResourceName:
+		return s.Resource
+	case ext.SpanType:
+		return s.Type
+	// Bool.
+	case ext.AnalyticsEvent:
+		return s.Metrics[ext.EventSampleRate] == 1.0
+	case ext.ManualDrop:
+		return s.Metrics[keySamplingPriority] == -1
+	case ext.ManualKeep:
+		return s.Metrics[keySamplingPriority] == 2
+	// Metrics.
+	case ext.SamplingPriority:
```

Comment:
```suggestion
	case ext.SamplingPriority, keySamplingPriority:
```
And you can remove the below